libs/cases/backend/case-management/src/lib/services/case.service.ts

```diff
@@ -90,6 +105,14 @@ export class XdCaseService {
 				...item,
 				overdue: Date.now() > new Date(item.dueDate).getTime(),
 			})),
+			catchError((error) => {
+				if (error instanceof Prisma.PrismaClientKnownRequestError) {
+					if (error.code === 'P2025') {
+						throw new NotFoundException(`Case with ID ${id} does not exist.`);
+					}
+				}
+				throw new ConflictException('An error occurred while deleting the case.');
```

Comment:
Pls be consitent with the Errors which are used. Why don't you use BadRequest Exception as well?